libfirm-rs/src/nodes_gen.rs

```diff
@@ -0,0 +1,2786 @@
+    fn create_mod(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Mod(Mod(ir_node))
+    }
+    fn create_mul(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Mul(Mul(ir_node))
+    }
+    fn create_mulh(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Mulh(Mulh(ir_node))
+    }
+    fn create_mux(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Mux(Mux(ir_node))
+    }
+    fn create_nomem(ir_node: *mut bindings::ir_node) -> Node {
+        Node::NoMem(NoMem(ir_node))
+    }
+    fn create_not(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Not(Not(ir_node))
+    }
+    fn create_offset(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Offset(Offset(ir_node))
+    }
+    fn create_or(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Or(Or(ir_node))
+    }
+    fn create_phi(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Phi(Phi(ir_node))
+    }
+    fn create_pin(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Pin(Pin(ir_node))
+    }
+    fn create_proj(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Proj(Proj(ir_node))
+    }
+    fn create_raise(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Raise(Raise(ir_node))
+    }
+    fn create_return(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Return(Return(ir_node))
+    }
+    fn create_sel(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Sel(Sel(ir_node))
+    }
+    fn create_shl(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Shl(Shl(ir_node))
+    }
+    fn create_shr(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Shr(Shr(ir_node))
+    }
+    fn create_shrs(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Shrs(Shrs(ir_node))
+    }
+    fn create_size(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Size(Size(ir_node))
+    }
+    fn create_start(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Start(Start(ir_node))
+    }
+    fn create_store(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Store(Store(ir_node))
+    }
+    fn create_sub(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Sub(Sub(ir_node))
+    }
+    fn create_switch(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Switch(Switch(ir_node))
+    }
+    fn create_sync(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Sync(Sync(ir_node))
+    }
+    fn create_tuple(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Tuple(Tuple(ir_node))
+    }
+    fn create_unknown(ir_node: *mut bindings::ir_node) -> Node {
+        Node::Unknown(Unknown(ir_node))
+    }
+}
+
+/// returns the sum of its operands
+#[derive(Debug)]
+pub struct Add(*mut bindings::ir_node);
+
+impl Add {
+    /// Gets first operand.
+    pub fn left(&self) -> Node {
+        let unwrapped = unsafe { bindings::get_Add_left(self.0) };
+        NodeFactory::node(unwrapped)
+    }
+
+    /// Sets first operand.
+    pub fn set_left(&self, val: &'_ Node) {
```

Comment:
Maybe an atomic 
```rust
pub fn set_left<F>(&self, F)
where F: FnOnce(Node) -> Node { /*...*/ }
```
is better, to reduce the danger of leaking pointers to about-to-be-replaced `left` arm

Reply:
I don't fully understand. What is the danger and how is it avoided by that?

Reply:
he is suggesting a replace function.